draft-ietf-quic-transport.md

```diff
@@ -2366,11 +2366,16 @@ genuine migrations.  Changing port number can cause a peer to reset its
 congestion state (see {{migration-cc}}), so the port SHOULD only be changed
 infrequently.
 
-Endpoints could also have their activity correlated if their peers keep using
-the same destination connection ID after migration. Nodes that initiate a
-migration need to provide their peers with new connection IDs. The goal is
+Endpoints that use connection IDs with length greater than zero
+could also have their activity correlated if their peers keep using
+the same destination connection ID after migration. If a node receives
+packets through a newly used connection ID, it SHOULD select an
```

Comment:
I'm wondering whether this should be a MUST in this case.  Not only are you migrating, but the peer has changed CIDs, which seems like strong evidence it's trying to limit linkability?

Reply:
Yes, it would seem rather pointerless to change CID otherwise. Responding with an old CID is the same as the peer never changing CID. The only reason is running out of CID's and not being that sensitive to linkability. As we now, opting out voluntarily does not work well in crypto.

Reply:
@mikkelfj the whole point of this PR is to say that and ensure that it works. Do you want to specify what the server should do if the client's probe arrives before a new connection ID is available? By your logic, we want to say that the server should wait before responding to the probe.

Then, do you want to enforce that? For example, suppose that a client sent a probe with a new CID, and receives a response with an already-in-use CID. Should it just shrug and continue, or signal a protocol violation?

Reply:
Responding on the old path is problematic too, because it causes correlation between new and old path.